### PR TITLE
Socint 427

### PR DESCRIPTION
**Pull request by KieranWardle**
https://collaborate2.ons.gov.uk/jira/browse/SOCINT-427

Added endpoint to get all case interactions for a given userIdentity in descending order

### CODE REVIEW

File: src/main/resources/db/migration/V21__ReadCaseInteractionPermission.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+INSERT INTO permission (id, role_id, permission_type)
```
Comment (rpcatling) [inline src/main/resources/db/migration/V21__ReadCaseInteractionPermission.sql:1]: need to add READ_USER_INTERACTIONS to super-user also

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/InteractionServiceImplTest.java [hunk 2]
```diff
@@ -115,6 +122,51 @@ public void shouldAcceptInteractionNotNeedingSubtype() throws CTPException {
...
+
+    when(interactionRepository.findAllByCcuserId(USER_ID)).thenReturn(interactions);
+
+    List<UsersCaseInteractionDTO> response =
+        interactionService.getAllCaseInteractionsForUser(USER_IDENTITY);
+
+    assertTrue(response.get(0).getCreatedDateTime().isAfter(response.get(1).getCreatedDateTime()));
+    assertTrue(response.get(1).getCreatedDateTime().isAfter(response.get(2).getCreatedDateTime()));
+
+    // TODO TEST retreiving caseInteractionas by caseId
```
Comment (rpcatling) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/InteractionServiceImplTest.java:146]: is this unfinished work ?

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/InteractionEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  private RBACService rbacService;
+
+  /**
+   * Constructor for ContactCentreDataEndpoint
+   *
+   * @param rbacService performs permission checking
+   * @param interactionService is a service layer object that we be doing the processing on behalf
+   *     of this endpoint.
+   */
+  @Autowired
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/InteractionEndpoint.java:41]: you can omit the `@Autowired` if you like with constructor injection

**Review by rpcatling — CHANGES_REQUESTED** [review]
good work; just a few comments

File: src/main/resources/db/migration/V21__ReadCaseInteractionPermission.sql (hunk 1 above)
Comment (philwhiles) [reply src/main/resources/db/migration/V21__ReadCaseInteractionPermission.sql:1]: Agreed

File: src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/InteractionEndpointIT.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.CollectionExerciseRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.RoleRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.SurveyRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.UserAuditRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.UserRepository;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.UsersCaseInteractionDTO;
+
+public class InteractionEndpointIT extends FullStackIntegrationTestBase {
+
+  private static final UUID TARGET_USER = UUID.fromString("4f27ee97-7ba7-4979-b9e8-bfe3063b41e8");
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/endpoint/InteractionEndpointIT.java:49]: would be better named 'principal' to keep in line with the terminology elsewhere in the code base.
The principal is always the user asking for the thing to be done, target makes it  sound as if something is being done to that user.

**Review by philwhiles — CHANGES_REQUESTED** [review]
see comments

File: src/main/resources/application-local.yml [hunk 5]
```diff
@@ -3,7 +3,7 @@ spring:
     gcp:
       pubsub:
         project-id: local
-        emulator-host: localhost:9808
+        emulator-host: localhost:8085
```
Comment (rpcatling) [inline src/main/resources/application-local.yml:6]: please revert the port number back to 9808

**Review by rpcatling — CHANGES_REQUESTED** [review]

**Review by rpcatling — APPROVED** [review]

**Pull request merged**